server/accounts.go

```diff
@@ -2647,6 +2647,10 @@ func (s *Server) UpdateAccountClaims(a *Account, ac *jwt.AccountClaims) {
 			if incomplete && name != old.Name {
 				if accClaims, _, err := s.verifyAccountClaims(claimJWT); err == nil {
 					s.UpdateAccountClaims(acc, accClaims)
+					// old.Name was deleted before ranging over accounts
```

Comment:
I am uneasy with this approach of doing it in place (as opposed to timer based). Say that we had A import from B that import from C, and all this chain is broken (meaning processing of A import failed, and B import failed).
Say that we are processing A at the moment (so old.Name == A). There is no guarantee of ordering doing the accounts.Range(). So say that the first we get is "B" in the Range() and it is incomplete (and name != A) then we recursively call this function. Now we are back at line 2636 and we have "B" marked as incomplete. We range and now say that the first in the Range() is "A". It is still incomplete and we now recursively call this function again, and we happen to going from "A" being given first in Range(), to "B", would that not end-up panic'ing with a stack overflow?

Reply:
permanently broken circular imports did the trick. fixed